picard/ui/options/cover.py

```diff
@@ -61,6 +61,10 @@ def __init__(self, parent=None):
     def load_cover_art_providers(self):
         """Load available providers, initialize provider-specific options, restore state of each
         """
+        # Remove current provider items during a reset
+        if self.ui.ca_providers_list.count() > 1:
+            self.ui.ca_providers_list.itemAt(0).widget().setParent(None)
```

Comment:
I'm not fond of this approach.
If i understand correctly you test for `self.ui.ca_providers_list.count() > 1` because `ca_providers_list` has an element added by https://github.com/suhas2go/picard/blob/fc36cb0a28c150c051182bda654557316a1cf195/picard/ui/ui_options_cover.py#L62-L65

The thing is this can be modified in https://github.com/suhas2go/picard/blob/fc36cb0a28c150c051182bda654557316a1cf195/ui/options_cover.ui#L80 and if an element is added, this code will break.

Perhaps a better approach would be to use our own Widget-derived class with a marker, and iterating over widgets, selecting marked ones, remove them. So this "reset" will not depend on anything else.